utils/merkletrie/fsnoder/file.go

```diff
@@ -0,0 +1,73 @@
+package fsnoder
+
+import (
+	"bytes"
+	"fmt"
+	"hash/fnv"
+
+	"gopkg.in/src-d/go-git.v4/utils/merkletrie/noder"
+)
+
+// file values represent file-like noders in a merkle trie.
+type file struct {
+	name     string // relative
+	contents string
+	hash     []byte // memoized
+}
+
+// newFile returns a noder representing a file with the given contents.
+func newFile(name, contents string) (*file, error) {
+	if name == "" {
+		return nil, fmt.Errorf("files cannot have empty names")
+	}
+
+	return &file{
+		name:     name,
+		contents: contents,
+	}, nil
+}
+
+// The hash of a file is just its contents.
+// Empty files will have the fnv64 basis offset as its hash.
+func (f *file) Hash() []byte {
+	if f.hash == nil {
+		h := fnv.New64a()
+		h.Write([]byte(f.contents))
```

Comment:
No, according to its description:
```
It never returns an error.
```
see https://golang.org/pkg/hash/#Hash for more information